apps/backend/src/opencage/opencage.controller.ts

```diff
@@ -0,0 +1,31 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import * as https from 'https';
+
+@Controller('api/v1/location')
+export class OpencageController {
+  @Get()
+  async reverseGeocoding(@Query('latitude') latitude: string, @Query('longitude') longitude: string): Promise<any> {
+    if (!latitude || !longitude) {
+      throw new Error('Latitude and longitude must be provided.');
+    }
+
+    const url = `https://api.opencagedata.com/geocode/v1/json?q=${latitude}+${longitude}&key=${process.env.OPENCAGE_KEY}`;
+
+    return new Promise((resolve, reject) => {
+      https.get(url, (response) => {
```

Comment:
nest has a built-in http module you could use instead https://docs.nestjs.com/techniques/http-module#http-module

Reply:
I am experiencing the following error when trying to use the `@nestjs/axios` package: "Cannot find module `@nestjs/axios' or its corresponding type declarations."

I tried several approaches to install and use `@nestjs/axios`, but I was unable to resolve the issue. I believe that my lack of experience with PNPM may be contributing to this.

If you could offer any guidance or suggestions on how to resolve this issue, I would greatly appreciate the help.

Reply:
Sure, I'll take a look and maybe we can hop on a call on discord and pair on it

Reply:
so looks like we have a problem on our docker configuration. every time we add a new dependency it's expected we have to rebuild the container because the `node_modules` is cached, so something like `docker-compose up --rebuild` should work. however, I only got this working when I manually deleted the container and ran the command again. I'll take a look at it later, but for now you can:

1. install the dependencies with `pnpm add -D @nestjs/axios axios --filter=backend`
2. delete the container manually
3. `docker-compose up`